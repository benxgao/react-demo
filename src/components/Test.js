import React from 'react';

export default class Test extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: 'hello'
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(() => ({
      text: 'mingxin'
    }));
  }

  render() {
    return (
      <div className="test-class"
        ref={(input) => { this.textInput = input; }}
        onClick={this.handleClick}>
        {this.state.text}
      </div>
    );
  }
}
