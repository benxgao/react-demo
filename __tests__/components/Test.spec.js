import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Test from '../../src/components/Test';

describe.skip('<Test />', () => {
  it('init state', () => {
    const wrapper = shallow(
      <Test />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('click', () => {
    const wrapper = shallow(
      <Test />
    );

    wrapper.find('.test-class').simulate('click');

    expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
