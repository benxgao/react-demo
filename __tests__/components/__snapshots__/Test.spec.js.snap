// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> should show default text when inital load 1`] = `
<div
  className="test-class"
  onClick={[Function]}
>
  hello
</div>
`;

exports[`<Test /> should show diff text when it is clicked 1`] = `
<div
  className="test-class"
  onClick={[Function]}
>
  mingxin
</div>
`;
